File: common/src/main/java/org/sandboxpowered/silica/network/clientbound/EncryptionRequest.java
package org.sandboxpowered.silica.network.clientbound;

import org.sandboxpowered.silica.network.Packet;
import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.server.SilicaServer;

public class EncryptionRequest implements Packet {
    private String serverId;
    private byte[] publicKey;
    private byte[] verifyArray;

    public EncryptionRequest() {
    }

    public EncryptionRequest(String serverId, byte[] publicKey, byte[] verifyArray) {
        this.serverId = serverId;
        this.publicKey = publicKey;
        this.verifyArray = verifyArray;
    }

    @Override
    public void read(PacketByteBuf buf) {
        serverId = buf.readString(20);
        publicKey = buf.readByteArray();
        verifyArray = buf.readByteArray();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeString(this.serverId);
        buf.writeByteArray(this.publicKey);
        buf.writeByteArray(this.verifyArray);
    }

    @Override
    public void handle(PacketHandler packetHandler, SilicaServer server) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/login/clientbound/EncryptionRequest.kt
package org.sandboxpowered.silica.network.login.clientbound

import org.sandboxpowered.silica.network.Connection
import org.sandboxpowered.silica.network.Packet
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler

class EncryptionRequest(
    private var serverId: String?,
    private var publicKey: ByteArray?,
    private var verifyArray: ByteArray?
) : Packet {

    constructor() : this(null, null, null)

    override fun read(buf: PacketByteBuf) {
        serverId = buf.readString(20)
        publicKey = buf.readByteArray()
        verifyArray = buf.readByteArray()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeString(serverId!!)
        buf.writeByteArray(publicKey!!)
        buf.writeByteArray(verifyArray!!)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {}
}